api/src/main/java/com/community/api/services/QualificationDetailsService.java

```diff
@@ -11,166 +14,218 @@
-        }
-        qualificationDetails.setQualification_id(examinationToAdd);
+        Integer qualificationToAdd = null;
+        qualificationToAdd = findQualificationId(qualificationDetails.getQualification_id(), qualifications);
+        qualificationDetails.setQualification_id(qualificationToAdd);
         qualificationDetails.setCustom_customer(customCustomer);
         customCustomer.getQualificationDetailsList().add(qualificationDetails);
         entityManager.persist(qualificationDetails);
         return qualificationDetails;
     }
 
-    public List<QualificationDetails> getQualificationDetailsByCustomerId(Long customCustomerId) throws  CustomerDoesNotExistsException,RuntimeException {
-        CustomCustomer customCustomer= entityManager.find(CustomCustomer.class,customCustomerId);
-        if(customCustomer==null)
-        {
-            throw new CustomerDoesNotExistsException("Customer does not exist with id "+ customCustomerId);
+    @Transactional
+    public List<Map<String, Object>> getQualificationDetailsByCustomerId(Long userId, String roleName) throws CustomerDoesNotExistsException, RuntimeException {
+        List<QualificationDetails> qualificationDetails;
+        if (roleName.equals(Constant.SERVICE_PROVIDER)) {
+            ServiceProviderEntity serviceProviderEntity = findServiceProviderById(userId);
+            qualificationDetails = serviceProviderEntity.getQualificationDetailsList();
+            return sharedUtilityService.mapQualifications(qualificationDetails);
         }
-       List<QualificationDetails> qualificationDetails = customCustomer.getQualificationDetailsList();
-        return qualificationDetails;
+        CustomCustomer customCustomer = findCustomCustomerById(userId);
+        qualificationDetails = customCustomer.getQualificationDetailsList();
+        return sharedUtilityService.mapQualifications(qualificationDetails);
     }
 
     @Transactional
-    public QualificationDetails deleteQualificationDetail(Long customCustomerId, Long qualificationId) throws EntityDoesNotExistsException, CustomerDoesNotExistsException {
-        CustomCustomer customCustomer= entityManager.find(CustomCustomer.class,customCustomerId);
-        if(customCustomer==null)
-        {
-            throw new CustomerDoesNotExistsException("Customer does not exist with id "+ customCustomerId);
-        }
-        List<QualificationDetails> qualificationDetails = customCustomer.getQualificationDetailsList();
-        QualificationDetails qualificationDetailsToDelete =null;
-        for(QualificationDetails qualificationDetails1 : qualificationDetails)
-        {
-            if(qualificationDetails1.getId()==qualificationId)
-            {
+    public QualificationDetails deleteQualificationDetail(Long userId, Long qualificationId, String roleName) throws EntityDoesNotExistsException, CustomerDoesNotExistsException {
+        List<QualificationDetails> qualificationDetails;
```

Comment:
here if someone enters string
give NumberFormatException exception

use this
Long id = Long.valueOf(customerId);

catch (NumberFormatException e) {
    return ResponseService.generateErrorResponse("Invalid customerId: expected a Long", HttpStatus.BAD_REQUEST);
}